rewrite-maven/src/main/java/org/openrewrite/maven/MavenSettings.java

```diff
@@ -44,6 +52,8 @@
 @Data
 @AllArgsConstructor
 public class MavenSettings {
+    private static HashMap<Integer, Optional<MavenSettings>> settingsFromEnvironmentCache = new HashMap<>();
```

Comment:
It looks like if a new execution context were passed to `parse()`, potentially with a completely different settings.xml that it would be ignored in favor of whatever was passed in first.

Why do we need this cache?